cmake/onnxruntime_providers.cmake

```diff
@@ -112,6 +112,70 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/../include/onnxruntime/core/providers/cp
 set_target_properties(onnxruntime_providers PROPERTIES LINKER_LANGUAGE CXX)
 set_target_properties(onnxruntime_providers PROPERTIES FOLDER "ONNXRuntime")
 
+
+if (onnxruntime_USE_MIMALLOC)
+  set(mimalloc_root_dir ${PROJECT_SOURCE_DIR}/external/mimalloc)
+  set(mimalloc_output_dir ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE})
+  set(mimalloc_wheel_dir ${mimalloc_output_dir}/onnxruntime/capi/)
+
+  add_definitions(
+    -DUSE_MIMALLOC=1 # used in ONNXRuntime
+    -DMI_OVERRIDE=ON) # used in building MiMalloc
+  include_directories(${mimalloc_root_dir}/include)
+
+  if(NOT IS_DIRECTORY ${mimalloc_wheel_dir})
+    file(MAKE_DIRECTORY ${mimalloc_wheel_dir})
+  endif()
+  
+  if (WIN32)
+    # The generic MiMalloc CMakeLists.txt project lacks 
+    # the needed hooks to override malloc at runtime on Windows
+    # so we fall back to the specially provided VS solutions (which
+    # do have those hooks)
+    set(mimalloc_output mimalloc-override)
+
+    if(NOT ${CMAKE_GENERATOR_PLATFORM} MATCHES "x64|Win32")
+      message(FATAL_ERROR "MiMalloc doesn't support ARM/ARM64 targets")
+    endif()
+    
+    set(vs_version "vs2019")
+    if (${CMAKE_GENERATOR} MATCHES "Visual Studio [1-5]+ [0-9]+")
+      set(vs_version "vs2017")
+    endif()
+    
+    set(mimalloc_config "Release")
+    if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
+      set(mimalloc_config, "Debug")
+    endif()
+
+    add_custom_command(OUTPUT ${mimalloc_output} COMMAND msbuild ${mimalloc_root_dir}/ide/${vs_version}/mimalloc-override.vcxproj 
```

Comment:
You may try to read the "WindowsSDKVersion" env var.
msbuild is not always in the PATH. if it is, usually WindowsSDKVersion is also set.